src/res_core.ml

```diff
@@ -963,15 +984,17 @@ let parseTemplateConstant ~prefix (p : Parser.t) =
   Parser.nextTemplateLiteralToken p;
   match p.token with
   | TemplateTail txt ->
-    Parser.next p;
+    Parser.next p [@doesNotRaise];
     let txt = if p.mode = ParseForTypeChecker then parseTemplateStringLiteral txt else txt in
     Parsetree.Pconst_string (txt, prefix)
   | _ ->
     let rec skipTokens () =
-      Parser.next p;
-      match p.token with
-      | Backtick -> Parser.next p; ()
-      | _ -> skipTokens ()
+      if p.token <> Eof then (
```

Comment:
And this is an infinite loop:
```res
let foo = x =>
  switch x {
  | `${
```

See https://github.com/rescript-lang/syntax/pull/542